src/definitions/modules/calendar.js

```diff
@@ -157,7 +157,7 @@ $.fn.calendar = function(parameters) {
             var onShow = function () {
               //reset the focus date onShow
               module.set.focusDate(module.get.date());
-              module.set.mode(settings.startMode);
+              module.set.mode(module.get.validatedMode(settings.startMode));
```

Comment:
Maybe stupid question.
Why not `module.set.mode(modue.get.mode())` here ?
`module.get.mode()` attempts to get `$module.data(metadata.mode)` in addition to `settings.startMode`.

Reply:
This is needed, because at `onShow` we have to make sure, that always the startingMode will be used. If we would use `module.get.mode()` here and a value at $module.data(metadata.mode) is indeed found _and_ you left the previous opening of the calendar by cancelling it in a different  (for example year selection) mode, then the calendar will popup with that last remembered mode, which is wrong.
I adjusted a jsfiddle to demonstrate this here https://jsfiddle.net/lubber/gzsxr8t1/5/

### Wrong Behavior
![calendarkeepstartmode_bad](https://user-images.githubusercontent.com/18379884/85258576-04e5f800-b468-11ea-9636-79f497f11765.gif)

### Correct behavior
![calendarkeepstartmode_good](https://user-images.githubusercontent.com/18379884/85258592-09aaac00-b468-11ea-9607-f15fed80babb.gif)

Reply:
Thanks for detailed explanation 👍 
That helps understanding current calendar module implementation.